### ISSUE TITLE
Mega Miner: Black screen

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#9956]
### Describe the bug

Link to SWF: http://i.notdoppler.com/files/megaminer.swf

[Mega Miner](http://www.notdoppler.com/megaminer.php) shows nothing but a black screen. As stated in https://github.com/ruffle-rs/ruffle/issues/1000#issuecomment-1459227097, the game needs TLF text support and `XML.settings()`.

### Expected behavior

The game should start.

### Affected platform

Desktop app

### Operating system

Windows 10

**Comment by n0samu** [comment]
Now the error is:
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "addedToStage", class: flash.events::Event, ptr: 0x1d33e2a3e80 }) to handler FunctionObject(FunctionObject { ptr: 0x1d362342050 }) : ReferenceError: Error #1069: Property perspectiveProjection not found on flash.geom.Transform and there is no default value.
        at MegaMiner_AG_fla::MainTimeline/__setPerspectiveProjection_()
```

**Comment by Lord-McSweeney** [comment]
Also `ReferenceError: Error #1069: Property http://adobe.com/AS3/2006/builtin::hasOwnProperty not found on Class and there is no default value.`

**Comment by n0samu** [comment]
With #10812, the new error is:
```
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "addedToStage", class: flash.events::Event, ptr: 0x1d3df471638 }) to handler FunctionObject(FunctionObject { ptr: 0x1d3df6817d8 }) : ReferenceError: Error #1069: Property perspectiveProjection not found on flash.geom.Transform and there is no default value.
        at MegaMiner_AG_fla::MainTimeline/__setPerspectiveProjection_()
```

**Comment by Lord-McSweeney** [comment]
This now spams out warnings repeatedly, still stuck on a black screen:
```
Display object container has no child with name __id19_
Display object container has no child with name __id1_
Display object container has no child with name __id19_
Display object container has no child with name __id1_
Display object container has no child with name __id19_
Display object container has no child with name __id1_
Display object container has no child with name __id19_
Display object container has no child with name __id1_
...
```

**Comment by n0samu** [comment]
I actually fixed this with #11523, I just didn't notice because the Play button is invisible. Clicking its area goes to the main menu, from which you can click the invisible menu buttons and start the game. Not great but it works!
